tests/test_api.rs

```diff
@@ -1004,6 +1004,15 @@ fn set_flags_from_command_line() {
   );
 }
 
+#[test]
+fn set_flags_from_command_line_with_usage() {
```

Comment:
This actually prints to stdout regardless if `--nocapture` is specified on the command line. If this feature is something that seems reasonable to add, then I'll see if I can work around that.